Add a tool to test generated datastreams

jan-cerny opened the pull request:
#### Description:
This script can compare 2 datastreams for presence of rules, OVALs and remediations, it can also show the diff between the remediation contents and also partially compares OVAL (only the criteria elements).

#### Rationale:
This script will help us to ensure no regression is made during the templating system redesign, but it might be useful also elsewhere.

matejak commented on utils/compare_ds.py:
```diff
@@ -0,0 +1,274 @@
+
+import ssg.constants
+
+ns = {
```
Couldn't we put this into constants?

matejak commented on utils/compare_ds.py:
```diff
@@ -0,0 +1,274 @@
+    "catalog": "urn:oasis:names:tc:entity:xmlns:xml:catalog",
+    "xlink": "http://www.w3.org/1999/xlink",
+}
+remediation_type_to_uri = {
```
ditto

matejak commented on utils/compare_ds.py:
```diff
@@ -0,0 +1,274 @@
+    parser = argparse.ArgumentParser(
+        description="Compares two datastreams with regards to presence of"
+        "OVAL checks and all remediations")
+    parser.add_argument(
```
Make the old and new DS filenames required positional arguments rather than required optional arguments. The current approach is somehow [non-standard](https://docs.python.org/3/library/argparse.html#required).

matejak commented on utils/compare_ds.py:
```diff
@@ -0,0 +1,274 @@
+        rules_in_old_benchmark = old_benchmark.findall(
+            ".//xccdf:Rule[@id='%s']" % (rule_id), ns)
+    else:
+        rules_in_old_benchmark = old_benchmark.findall(".//xccdf:Rule", ns)
```
Ideally, split this into "get_rules_to_compare" and "compare_rules" functions - the original one is quite big and function names act as comments pointing towards what's going on.

matejak commented on utils/compare_ds.py:
```diff
@@ -0,0 +1,274 @@
+    elif (old_oval_ref is not None and new_oval_ref is None):
+        print("New datastream is missing OVAL for rule '%s'." % (rule_id))
+    elif (old_oval_ref is not None and new_oval_ref is not None):
+        old_check_content_ref = old_oval_ref.find(
```
Please extract this into a separate function with a good name, we have some serious `if` nesting here.

jan-cerny commented:
@matejak I have used constants from ssg.constants module whenever possible, refactored the code in order to extract functions `get_rules_to_compare` and `compare_rules`, then I have refactored the code to extract function `compare_fix_elements`. I have also renamed function `compare_remediation_contents` to `compare_fix_texts`. I have made `--old` and `--new` optional arguments.

matejak commented:
Good job! There is some confusion in the reply - the `old` and `new` arguments were converted to positional and they are required, as it was suggested. The new code is also easier to understand, which is great.

I have thought that you will put the whole namespace dict into the constant module as it is very userful in general, but this is also fine, and it can be moved later when conditions are even better.

Thank you for the PR!
